source/server/server.cc

```diff
@@ -249,27 +250,9 @@ void InstanceImpl::initialize(const Options& options,
             restarter_.version());
 
   ENVOY_LOG(info, "statically linked extensions:");
-  ENVOY_LOG(info, "  access_loggers: {}",
-            Registry::FactoryRegistry<Configuration::AccessLogInstanceFactory>::allFactoryNames());
-  ENVOY_LOG(
-      info, "  filters.http: {}",
-      Registry::FactoryRegistry<Configuration::NamedHttpFilterConfigFactory>::allFactoryNames());
-  ENVOY_LOG(info, "  filters.listener: {}",
-            Registry::FactoryRegistry<
-                Configuration::NamedListenerFilterConfigFactory>::allFactoryNames());
-  ENVOY_LOG(
-      info, "  filters.network: {}",
-      Registry::FactoryRegistry<Configuration::NamedNetworkFilterConfigFactory>::allFactoryNames());
-  ENVOY_LOG(info, "  stat_sinks: {}",
-            Registry::FactoryRegistry<Configuration::StatsSinkFactory>::allFactoryNames());
-  ENVOY_LOG(info, "  tracers: {}",
-            Registry::FactoryRegistry<Configuration::TracerFactory>::allFactoryNames());
-  ENVOY_LOG(info, "  transport_sockets.downstream: {}",
-            Registry::FactoryRegistry<
-                Configuration::DownstreamTransportSocketConfigFactory>::allFactoryNames());
-  ENVOY_LOG(info, "  transport_sockets.upstream: {}",
-            Registry::FactoryRegistry<
-                Configuration::UpstreamTransportSocketConfigFactory>::allFactoryNames());
+  for (const auto& ext : Envoy::Registry::FactoryCategoryRegistry::factories()) {
```

Comment:
Can you paste the output of this after this change (and the sorting comment)? It would be good to identify if anything will be confusing to end users.

Reply:
```
[2019-11-06 16:23:41.051][22677834][info][main] [source/server/server.cc:252] statically linked extensions:
[2019-11-06 16:23:41.051][22677834][info][main] [source/server/server.cc:254]   access_loggers: envoy.file_access_log, envoy.http_grpc_access_log, envoy.tcp_grpc_access_log
[2019-11-06 16:23:41.051][22677834][info][main] [source/server/server.cc:254]   clusters: envoy.cluster.eds, envoy.cluster.logical_dns, envoy.cluster.original_dst, envoy.cluster.static, envoy.cluster.strict_dns, envoy.clusters.dynamic_forward_proxy, envoy.clusters.redis
[2019-11-06 16:23:41.051][22677834][info][main] [source/server/server.cc:254]   dubbo_proxy.filters: envoy.filters.dubbo.router
[2019-11-06 16:23:41.051][22677834][info][main] [source/server/server.cc:254]   dubbo_proxy.protocols: dubbo
[2019-11-06 16:23:41.051][22677834][info][main] [source/server/server.cc:254]   dubbo_proxy.route_matchers: default
[2019-11-06 16:23:41.051][22677834][info][main] [source/server/server.cc:254]   dubbo_proxy.serializers: dubbo.hessian2
[2019-11-06 16:23:41.051][22677834][info][main] [source/server/server.cc:254]   filters.http: envoy.buffer, envoy.cors, envoy.csrf, envoy.ext_authz, envoy.fault, envoy.filters.http.adaptive_concurrency, envoy.filters.http.dynamic_forward_proxy, envoy.filters.http.grpc_http1_reverse_bridge, envoy.filters.http.grpc_stats, envoy.filters.http.header_to_metadata, envoy.filters.http.jwt_authn, envoy.filters.http.original_src, envoy.filters.http.rbac, envoy.filters.http.tap, envoy.grpc_http1_bridge, envoy.grpc_json_transcoder, envoy.grpc_web, envoy.gzip, envoy.health_check, envoy.http_dynamo_filter, envoy.ip_tagging, envoy.lua, envoy.rate_limit, envoy.router, envoy.squash
[2019-11-06 16:23:41.051][22677834][info][main] [source/server/server.cc:254]   filters.listener: envoy.listener.http_inspector, envoy.listener.original_dst, envoy.listener.original_src, envoy.listener.proxy_protocol, envoy.listener.tls_inspector
[2019-11-06 16:23:41.051][22677834][info][main] [source/server/server.cc:254]   filters.network: envoy.client_ssl_auth, envoy.echo, envoy.ext_authz, envoy.filters.network.dubbo_proxy, envoy.filters.network.mysql_proxy, envoy.filters.network.rbac, envoy.filters.network.sni_cluster, envoy.filters.network.thrift_proxy, envoy.filters.network.zookeeper_proxy, envoy.http_connection_manager, envoy.mongo_proxy, envoy.ratelimit, envoy.redis_proxy, envoy.tcp_proxy
[2019-11-06 16:23:41.051][22677834][info][main] [source/server/server.cc:254]   grpc_credentials: envoy.grpc_credentials.aws_iam, envoy.grpc_credentials.default, envoy.grpc_credentials.file_based_metadata
[2019-11-06 16:23:41.051][22677834][info][main] [source/server/server.cc:254]   health_checkers: envoy.health_checkers.redis
[2019-11-06 16:23:41.051][22677834][info][main] [source/server/server.cc:254]   resolvers: envoy.ip
[2019-11-06 16:23:41.051][22677834][info][main] [source/server/server.cc:254]   resource_monitors: envoy.resource_monitors.fixed_heap, envoy.resource_monitors.injected_resource
[2019-11-06 16:23:41.052][22677834][info][main] [source/server/server.cc:254]   retry_host_predicates: envoy.retry_host_predicates.omit_canary_hosts, envoy.retry_host_predicates.previous_hosts
[2019-11-06 16:23:41.052][22677834][info][main] [source/server/server.cc:254]   retry_priorities: envoy.retry_priorities.previous_priorities
[2019-11-06 16:23:41.052][22677834][info][main] [source/server/server.cc:254]   singletons: alts_shared_state_singleton, date_provider_singleton, dns_cache_manager_singleton, grpc_access_logger_cache_singleton, redis_redirection_manager_singleton, route_config_provider_manager_singleton, scoped_routes_config_provider_manager_singleton, tap_admin_handler_singleton
[2019-11-06 16:23:41.052][22677834][info][main] [source/server/server.cc:254]   ssl_contexts: ssl_context_manager
[2019-11-06 16:23:41.052][22677834][info][main] [source/server/server.cc:254]   stats_sinks: envoy.dog_statsd, envoy.metrics_service, envoy.stat_sinks.hystrix, envoy.statsd
[2019-11-06 16:23:41.052][22677834][info][main] [source/server/server.cc:254]   thrift_proxy.filters: envoy.filters.thrift.rate_limit, envoy.filters.thrift.router
[2019-11-06 16:23:41.052][22677834][info][main] [source/server/server.cc:254]   thrift_proxy.protocols: auto, binary, binary/non-strict, compact, twitter
[2019-11-06 16:23:41.052][22677834][info][main] [source/server/server.cc:254]   thrift_proxy.transports: auto, framed, header, unframed
[2019-11-06 16:23:41.052][22677834][info][main] [source/server/server.cc:254]   tracers: envoy.dynamic.ot, envoy.lightstep, envoy.tracers.datadog, envoy.tracers.opencensus, envoy.tracers.xray, envoy.zipkin
[2019-11-06 16:23:41.052][22677834][info][main] [source/server/server.cc:254]   transport_sockets.downstream: envoy.transport_sockets.alts, envoy.transport_sockets.raw_buffer, envoy.transport_sockets.tap, envoy.transport_sockets.tls, raw_buffer, tls
[2019-11-06 16:23:41.052][22677834][info][main] [source/server/server.cc:254]   transport_sockets.upstream: envoy.transport_sockets.alts, envoy.transport_sockets.raw_buffer, envoy.transport_sockets.tap, envoy.transport_sockets.tls, raw_buffer, tls
```

Reply:
Note that UDP factories aren't shown here (maybe I need to enable some build config?). You might want to verify that I got reasonable category names for `ActiveUdpListenerConfigFactory` and `NamedUdpListenerFilterConfigFactory`.

Reply:
I'm working on a bunch of UDP stuff and I can check this locally.

In general this is amazing. Thanks so much for doing this. My main comment is can we figure out how to hide `singletons` and `ssl_contexts` for now (or at least show them at debug log level vs. info). I think this isn't going to mean anything to users. 

Maybe you could just have some boolean property on a category that would say whether to hide it from the returned categories to be printed? WDYT?

Reply:
I added a get-out-of-jail-free card for these special cases.